lib/vimaly/client.rb

```diff
@@ -54,13 +54,24 @@ def update_ticket(id, other_fields)
       end
     end
 
-    def search_tickets(query_string, state: 'active')
+    def search_tickets(query_string, state: 'active', max_results: 500)
       ticket_type_map = Hash[ticket_types.map { |t| [t.id, t] }]
       bin_map = Hash[bins.map { |b| [b.id, b] }]
 
-      get("/ticket-search?text=#{query_string}&state=#{state}").map do |ticket_data|
-        Ticket.from_vimaly(ticket_data, ticket_type_map, bin_map, custom_fields)
+      response_data = []
+      i = 0
+
+      while (i * max_results) <= 5000
```

Comment:
I think this loop should stop right away once we don't get back the limit of `500` results in order to save on number of requests and time. Have a look at the `bins` method, the logic is a bit complex but does what it should do. And maybe add a test for this?

Reply:
I've rewritten this loop. 
It is quite different, for 2 reasons.

It will now stop straight away when we dont get the limit of 500 results back. 

And to account for when only 1 ticket gets returned per response (which I made possible to make it easier to test)

Reply:
It is rather complex. I've tried to write some comments in there. Let me know if they're enough